### ISSUE TITLE
Contributor ladder - draft data for review

### ISSUE THREAD

**Issue by juliana-hsu** [open-cluster-management-io/community#53]
Please provide comments on Contributor ladder template - draft data here:  https://docs.google.com/document/d/1-zBXqvnGaxPPVHbcoEyyzlkK0UI3eU1RXVQYpS2dVmU/edit
Ignore preliminary draft and reference new one below.

**Comment by jberkus** [comment]
So, I'm working on a draft of this, and to help this along I'd like to ask some questions of @qiujian16 and @mikeshng to fill in some bits:

1.  Will we have "subproject owners"?  Or are all maintainers part of one big pool?  Example: does the Policy Manager have its own owner?
2. OWNERS files seem to be pre-repo, with no OWNERs for specific directories/libraries.  Is this correct throughout the project?
3. Do we have any plans to have github management tools, e.g. Peribolous?  Or should I assume that all such activity (such as adding people to teams, etc.) is 100% manual?
4. Does our GH-actions based automation for CI/CD understand Approvers vs. Reviewers in the OWNERS files?  That is, what actions are Approvers able to take with automation, and what actions can Reviewers take, or is that controlled elsewhere (such as by GH Team membership?)
5. My suggestion for initial governance for the project is a simple Maintainer Council, where all maintainers get a vote. Is there any reason to do something else?

**Comment by qiujian16** [comment]
FYI @jberkus 

> So, I'm working on a draft of this, and to help this along I'd like to ask some questions of @qiujian16 and @mikeshng to fill in some bits:
> 
> 1. Will we have "subproject owners"?  Or are all maintainers part of one big pool?  Example: does the Policy Manager have its own owner?

Yes, we have "subproject owners"

> 2. OWNERS files seem to be pre-repo, with no OWNERs for specific directories/libraries.  Is this correct throughout the project?

That is  correct

> 3. Do we have any plans to have github management tools, e.g. Peribolous?  Or should I assume that all such activity (such as adding people to teams, etc.) is 100% manual?

We do not have this plan yet, currently it is 100% manual.

> 4. Does our GH-actions based automation for CI/CD understand Approvers vs. Reviewers in the OWNERS files?  That is, what actions are Approvers able to take with automation, and what actions can Reviewers take, or is that controlled elsewhere (such as by GH Team membership?)

That is still controlled by ci bot in openshift prow. Github action is only used for testing automation. merge and label management is done by ci bot in openshift prow.

> 5. My suggestion for initial governance for the project is a simple Maintainer Council, where all maintainers get a vote. Is there any reason to do something else?

Since the maintainer of each subproject will be different, should we need to define the Maintainer of the project and the maintainer of the subproject?

**Comment by jberkus** [comment]
> Since the maintainer of each subproject will be different, should we need to define the Maintainer of the project and the maintainer of the subproject?

We can completely do that, and I even have a template for it (thanks to Konveyor).  It's just a more complicated structure.  Give me a bit.

**Comment by qiujian16** [comment]
@mdelder any comment on this?

**Comment by qiujian16** [comment]
@jberkus I think you meant this https://github.com/konveyor/community/pull/8 in Konveyor?

**Comment by jberkus** [comment]
Yes.  I'm using Konveyor's draft ladder/governance for OCM's, since they are definitely a "project of projects".  You can take a look at their stuff to see what that would look like; some of the specifics will be different for OCM, but it's the same CNCF-created template.
